### DISCUSSION TITLE
How to do migration when index added

### DISCUSSION THREAD

**Discussion by HasanAlqaisi** [simolus3/drift#2766] [Q&A] [answered]
Hello,

I've added an index to a table using the `@TableIndex`. But I am not sure how to do the migration since the table already exists.
Note that I am adding the index to a new column as well, if that makes any difference.

**Comment by simolus3** [comment, marked as answer]
You can create the index separately in the migration. Drift will generate a getter in the database class for indices added to tables based on the name in the `TableIndex` annotation. So you can do

```dart
onUpgrade: (m, from, to) async {
  if (...) {
    await m.addColumn(table, table.newColumn);
    await m.create(tableIndex);
  }
}
```

There's been a bug fixed in drift\_dev 2.14.0 causing the getter to not be added in some cases - if you don't have it in your database, please try upgrading.
